CMakeLists.txt

```diff
@@ -47,6 +47,8 @@ ENDIF()
 INCLUDE(cmake/fake_tribits.cmake)
 INCLUDE(cmake/kokkoskernels_tribits.cmake)
 
+OPTION(BUILD_SHARED_LIBS "Build shared libraries" OFF)
```

Comment:
This will change the default build behavior of Kokkos Kernels which does a shared library build by default.
Is this change required for your fix?

Reply:
Oh, really? In fact, this patch requires building with shared libraries, see https://github.com/masterleinad/kokkos-kernels/blob/5a8e0146189c898551662b0ac9b1e2c580bfaa14/src/CMakeLists.txt#L470. I can check tomorrow if the on;y problem was that `Kokkos` didn't get built as a shared library.

Reply:
Actually after double checking, we are building static libraries by default so this line should not change anything.